File: exercism/python/space-age/space_age.py
class SpaceAge(object):
    def __init__(self, seconds):
        self.seconds = seconds

    def calculate_age(self, orbital_period):
        earth_years = self.seconds / 31557600
        age = earth_years / orbital_period
        return round(age, 2)

    def on_mercury(self):
        return self.calculate_age(0.2408467)

    def on_venus(self):
        return self.calculate_age(0.61519726)

    def on_earth(self):
        return self.calculate_age(1)

    def on_mars(self):
        return self.calculate_age(1.8808158)

    def on_jupiter(self):
        return self.calculate_age(11.862615)

    def on_saturn(self):
        return self.calculate_age(29.447498)

    def on_uranus(self):
        return self.calculate_age(84.016846)

    def on_neptune(self):
        return self.calculate_age(164.79132)

File: exercism/csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private int age;
    static private double earthYear = 31557600;
    static private double mercuryYear = earthYear * 0.2408467;
    static private double venusYear = earthYear * 0.61519726;
    static private double marsYear = earthYear * 1.8808158;
    static private double jupiterYear = earthYear * 11.862615;
    static private double saturnYear = earthYear * 29.447498;
    static private double uranusYear = earthYear * 84.016846;
    static private double neptuneYear = earthYear * 164.79132;

    public SpaceAge(int ageInSeconds)
    {
        age = ageInSeconds;
    }

    public double OnEarth() => age / earthYear;

    public double OnMercury() => age / mercuryYear;

    public double OnVenus() => age / venusYear;

    public double OnMars() => age / marsYear;

    public double OnJupiter() => age / jupiterYear;

    public double OnSaturn() => age / saturnYear;

    public double OnUranus() => age / uranusYear;

    public double OnNeptune() => age / neptuneYear;
}
